packages/search-input/src/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,377 @@
+    const handleSearchBoxFocus: FocusEventHandler = e => {
+      const target =
+        e.target !== clearButtonRef.current
+          ? inputRef.current ?? (e.target as HTMLElement)
+          : clearButtonRef.current;
+      target.focus();
+      trackFocusedElement(target);
+      handleOpenMenuAction(e);
+    };
+
+    const handleClearButton: MouseEventHandler<HTMLButtonElement> = e => {
+      // We convert the click event into a Change event,
+      // Update the target & value,
+      // Then fire any `onChange` handler
+      const changeEvent: ChangeEvent<any> = {
+        ...cloneDeep(e),
+        type: 'change',
+        target: inputRef.current as EventTarget,
+      };
+      changeEvent.target.value = '';
+      handleChange(changeEvent);
+      inputRef?.current?.focus();
+    };
+
+    const handleKeyDown: KeyboardEventHandler = e => {
+      const isFocusInMenu = menuRef.current?.contains(document.activeElement);
+      const isFocusOnSearchBox = searchBoxRef.current?.contains(
+        document.activeElement,
+      );
+
+      const isFocusInComponent = isFocusOnSearchBox || isFocusInMenu;
+
+      if (isFocusInComponent) {
+        switch (e.keyCode) {
+          case keyMap.Enter: {
+            const highlightedElementRef = resultRefs(`${highlightIndex}`);
+            highlightedElementRef?.current?.click();
+            break;
+          }
+
+          case keyMap.Escape: {
+            closeMenu();
+            inputRef.current?.focus();
+            break;
+          }
+
+          case keyMap.ArrowDown: {
+            if (withTypeAhead) {
+              inputRef.current?.focus();
+              e.preventDefault(); // Stop page scroll
+              updateHighlight('next');
+            }
+            break;
+          }
+
+          case keyMap.ArrowUp: {
+            if (withTypeAhead) {
+              inputRef.current?.focus();
+              e.preventDefault(); // Stop page scroll
+              updateHighlight('prev');
+            }
+            break;
+          }
+        }
+      }
+    };
+
+    const handleBlur: FocusEventHandler = e => {
+      if (
+        !isElementInComponent(e.target) &&
+        !isElementInComponent(document.activeElement)
+      ) {
+        closeMenu();
+      }
+    };
+
+    const handleSubmit: FormEventHandler<HTMLFormElement> = e => {
+      if (withTypeAhead) {
+        e.preventDefault();
+      } else {
+        onSubmitProp?.(e);
+      }
+    };
+
+    useBackdropClick(closeMenu, [searchBoxRef, menuRef], isOpen);
+
+    return (
+      <LeafyGreenProvider darkMode={darkMode}>
+        <SearchInputContextProvider
+          state={state}
+          highlight={highlightIndex}
+          resultDynamicRefs={resultRefs}
+        >
+          <form
+            role="search"
+            ref={formRef}
+            className={cx(formStyle, className)}
+            onSubmit={handleSubmit}
+            onBlur={handleBlur}
+            {...rest}
```

Comment:
should this have `rest[aria-label]` and `rest[aria-labelledby]` spread here?

Reply:
Good catch, I'll pull it out